sdk/communication/azure-communication-rooms/src/main/java/com/azure/communication/rooms/models/RoomParticipant.java

```diff
@@ -16,56 +15,53 @@ public final class RoomParticipant {
      * model must be interpreted as a union: Apart from rawId, at most one
      * further property may be set.
      */
-    @JsonProperty(value = "communicationIdentifier", required = true)
-    private CommunicationIdentifier communicationIdentifier;
+    private final CommunicationIdentifier communicationIdentifier;
 
     /*
      * The Role of a room participant.
      */
-    @JsonProperty(value = "role")
-    private RoleType role;
+    private final ParticipantRole role;
 
     /**
-     * Get the communicationIdentifier property: Identifies a participant in Azure Communication services. A participant
-     * is, for example, a phone number or an Azure communication user. This model must be interpreted as a union: Apart
-     * from rawId, at most one further property may be set.
+     * Default constructor for Room Participant. Default role is Attendee
      *
-     * @return the communicationIdentifier value.
+     * @param communicationIdentifier The communication identifier.
      */
-    public CommunicationIdentifier getCommunicationIdentifier() {
-        return this.communicationIdentifier;
+    public RoomParticipant(CommunicationIdentifier communicationIdentifier) {
+        this.communicationIdentifier = communicationIdentifier;
+        this.role = ParticipantRole.ATTENDEE;
     }
 
     /**
-     * Set the communicationIdentifier property: Identifies a participant in Azure Communication services. A participant
-     * is, for example, a phone number or an Azure communication user. This model must be interpreted as a union: Apart
-     * from rawId, at most one further property may be set.
+     * Constructor for Room Participant
      *
-     * @param communicationIdentifier the communicationIdentifier value to set.
-     * @return the RoomParticipant object itself.
+     * @param communicationIdentifier The communication identifier.
+     * @param role The participant role.
      */
-    public  RoomParticipant setCommunicationIdentifier(CommunicationIdentifier communicationIdentifier) {
+    public RoomParticipant(CommunicationIdentifier communicationIdentifier, ParticipantRole role) {
         this.communicationIdentifier = communicationIdentifier;
-        return this;
+        this.role = role;
```

Comment:
We should have a null check here, if the role is null, then set as `ParticipantRole.ATTENDEE`

Reply:
Won't need null check since one, as suggested by ARB, we will have a setter for role.

Reply:
Added